### PR TITLE
Update TAP information for jck tests

### LINKED ISSUE #439: Update jck tap file content to support enable jck_custom rerun link
If diagnostic=all, keep all outputs
if diagnostic=failure, filter the testcase and summarize result, same format as openjdk. This will

1.  make it easy to know which test cases failed
2. enable rerun jck_custom link with failed test cases.

### PR DESCRIPTION
**Pull request by sophia-guo**
Update  failed test target output parse in tap file to avoid the information cut and missing
Update jck test cases failed information to be the same way as openjdk tests for diagnostic=failure

Close #439

### CODE REVIEW

**Comment by sophia-guo** [comment]
Grinder:
temurin-compliance/job/Grinder/4087/

**Review by smlambert — APPROVED** [review]
thanks @sophia-guo !

**Comment by sophia-guo** [comment]
Update won't affect jdk tap files Grinder for jdk_custom still works fine https://ci.adoptium.net/job/Grinder/8992/
Grinder triggered by rerun link works fine https://ci.adoptium.net/job/Grinder/8993/

File: scripts/resultsSum.pl
Comment (sophia-guo) [reply scripts/resultsSum.pl:162]: See line 181.

Comment (llxia) [reply scripts/resultsSum.pl:162]: Can we change the logic to the following?
```
if (($diagnostic eq 'failure') || ($diagnostic eq 'all')) {
    if ($buildList =~ /jck/) {...}
    elsif ($buildList =~ /openjdk/) {...}
    else {..}
}
```

Comment (sophia-guo) [reply scripts/resultsSum.pl:162]: This is to parse, filter and record the failed test case name to a formatted tap file. Only make sense for diagnotic=failure. No need to do for diagnostic=noDetail and dianostic=all

Comment (llxia) [reply scripts/resultsSum.pl:162]: sure, we do not need to check for dianostic=all. The current if else block at line 181 is confusing.

Comment (sophia-guo) [reply scripts/resultsSum.pl:162]: Yes, the negation is normally difficult to understand. However, the way you mentioned would make the majority of code (current else block) confusing because extra if else would be needed.

Comment (sophia-guo) [reply scripts/resultsSum.pl:162]: I can update to 

```
if (($diagnostic eq 'failure') && ($buildList =~ /jck/ || $buildList =~ /openjdk/)) {
} else {
}
```

The difference would be the longer code block under the `if` branch and the one-liners of an exception move to `else` branch.

**Comment by smlambert** [comment]
Seems like this review is being held up over differing preferences for code style and readability.  

Given the related aqa-tests PR is merged (before this PR which it depends on), can we proceed to bring this in and update style later?  I am worried to hit a bunch of test jobs failing tonight otherwise.  ?  

I am also, waiting on this to merge before creating tags and release branch.

**Comment by llxia** [comment]
sigh, I did not notice https://github.com/adoptium/aqa-tests/pull/5106 was merged despite there being the `Depends on` message in the description.

I merge this in to unblock other activities.

**Review by llxia — APPROVED** [review]

**Comment by smlambert** [comment]
thanks @llxia !  I will proceed with https://github.com/adoptium/aqa-tests/issues/5103#issuecomment-1970331613 now.

**Pull request merged**
